viewing_party/party.py

```diff
@@ -1,23 +1,291 @@
+        # add movie into a new list if it is not duplicate and return that new list at the end
+        if movie not in friends_watched_list and movie not in user_unique_watched:
+            user_unique_watched.append(movie)
+    return user_unique_watched
+
+    
+def get_friends_unique_watched(user_data):      
+    """ Finding the list of movies that user has no watched but user's friend had watched at least one then return it. """
+
+    # assign user's friend watch list that is returning from get_friends_watched_movie function
+    friends_watched_list = get_friends_watched_movie(user_data)
+
+    # list of dictionary that store all movies are not in user watched list
+    friend_unique_watched = []
+
+    # compare user watched list and friends watched list
+    # loop to get each movie dictionary of firend watched movies list for comparing with user watched list          
+    for movie in friends_watched_list:
+
+        # check if the movie that at least one of their friend have watched but the user has not watched
+        # add that movie into a new list if it is not duplicate and return that new list at the end
+        if movie not in user_data["watched"] and movie not in friend_unique_watched:
+            friend_unique_watched.append(movie) 
+    return friend_unique_watched
+
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+
+def get_available_recs(user_data):
+    """ 
+    Finding recommended movies if:
+        - user has no watched
+        - at least one of user's friend watched it
+        - the "host" that is in "subscriptons" field
+        - return the list of recommended movies at the end 
+    """
+    # assign user's friend watch list that is returning from get_friends_watched_movie function
+    friend_watched_list = get_friends_watched_movie(user_data)
+    # list of movies that will be recommended when it matche all the conditions
+    recommend_movie_list = []
+
+    # return [] if friend watched movies is empty
+    if len(friend_watched_list) == 0 or len(user_data["watched"]) == 0:
+        return []
+
+    # otherwise, find recommended movies list
+    else: 
+        # loop over each movie in friend watched list to check the conditions
+        for movie in friend_watched_list:
+            # add movie into recommended movies list without duplicating if it matches the conditions as below
+            # check if user has not watched that movie and the host is in subcriptions
+            if movie not in user_data["watched"] and movie["host"] in user_data["subscriptions"]:    
+                # check if at least one of their friend have wathced and it is not in recommended movie list to avoid duplicated.
+                if friend_watched_list.count(movie) >= 1 and movie not in recommend_movie_list:
+                    recommend_movie_list.append(movie)

```

Comment:
The logic here is great! Consider splitting things onto multiple lines when they become this long so it's easier to read.